src/NuGet.Indexing/PackageMetadataExtraction.cs

```diff
@@ -110,39 +114,24 @@ static bool TryLoad(Stream stream, out XDocument document, List<string> errors)
             }
         }
 
-        static void ExtractId(IDictionary<string, string> package, XDocument document, List<string> errors)
-        {
-            XElement idElement = document.Root.DescendantsAndSelf().Elements().Where(d => d.Name.LocalName == "id").FirstOrDefault();
-            if (idElement != null)
-            {
-                package["id"] = (idElement.Value);
-            }
-            else
-            {
-                errors.Add("unable to find the id element in the nuspec");
-            }
-        }
-
-        static void ExtractVersion(IDictionary<string, string> package, XDocument document, List<string> errors)
+        static void ExtractRequiredProperty(IDictionary<string, string> package, XDocument document, List<string> errors, string localName, string key = null)
         {
-            XElement idElement = document.Root.DescendantsAndSelf().Elements().Where(d => d.Name.LocalName == "version").FirstOrDefault();
-            if (idElement != null)
-            {
-                package["version"] = (idElement.Value);
-            }
-            else
+            if (!ExtractProperty(package, document, localName, key))
             {
-                errors.Add("unable to find the version element in the nuspec");
+                errors.Add($"unable to find the '{localName}' element in the nuspec");
             }
         }
 
-        static void ExtractProperty(IDictionary<string, string> package, XDocument document, string name)
+        static bool ExtractProperty(IDictionary<string, string> package, XDocument document, string localName, string key = null)
```

Comment:
this is kind of a weird API, if you really want to keep it, please add an doc explaining the usage.

Also in general you want to avoid methods with default parameters values